File: Number_of_employees_who_meet_the_target.py
from typing import List 

class Solution:
    def numberOfEmployeesWhoMetTarget(self, hours: List[int], target: int) -> int:
        if not isinstance(hours, list) or not isinstance(target, int):
            raise TypeError('hours must be a valid list object and target must be valid int object')

        for elements in hours:
            if not isinstance(elements, int):
                raise TypeError('list elements must be valid int object type')
        
        if not hours or not target:
            return None
        
        count = 0
        for targets in range(len(hours)):
            if hours[targets] >= target:
                count += 1
        
        return count
    

obj = Solution()
print(obj.numberOfEmployeesWhoMetTarget([0,1,2,3,4], 2))
print(obj.numberOfEmployeesWhoMetTarget([5,1,4,2,2], 6))

File: same_solution_with_different_language/Number_of_employees_who_meet_the_target.js
/**
 * @param {number[]} hours
 * @param {number} target
 * @return {number}
 */
var numberOfEmployeesWhoMetTarget = function(hours, target) {
    if (!Array.isArray(hours) && !Number.isInteger(target)){
        throw new TypeError('hours must be valid Array and target must be valid Number');
    }
    for (let i = 0; i < hours.length; i++) {
        if (!Number.isInteger(hours[i])){
            throw new TypeError('Array elements must be valid Number');
        }
    }

    if (!hours) {
        return null;
    }

    let count  = 0;
    for (let j = 0; j < hours.length; j++) {
        if (hours[j] >= target){
            count += 1;
        }
    }

    return count;


};

console.log(numberOfEmployeesWhoMetTarget([0,1,2,3,4], 2));
console.log(numberOfEmployeesWhoMetTarget([5,1,4,2,2], 6));